src/as-metadata.c

```diff
@@ -100,8 +106,6 @@ as_metadata_file_guess_style (const gchar *filename)
 		return AS_FORMAT_STYLE_METAINFO;
 	if (g_str_has_suffix (filename, ".metainfo.xml.in.in"))
 		return AS_FORMAT_STYLE_METAINFO;
-	if (g_str_has_suffix (filename, ".xml"))
```

Comment:
Moving this looks wrong - if a pure match on `.xml` is executed first, the code that's checking if the suffix is e.g. `.metainfo.xml` will never be reached.

Reply:
Ouch, you are right. I wanted to have them together, not split (one at the top, another at the bottom) and I did not realize this. I moved the one from the top to the bottom instead.

Reply:
The set of the `if`-s is also kinda funny and slightly harder to read. Maybe this would be prettier? 
```c
	if (g_str_has_suffix (filename, ".xml") ||
	    g_str_has_suffix (filename, ".xml.gz") ||
	    g_str_has_suffix (filename, ".xml.zst"))
		return AS_FORMAT_STYLE_CATALOG;
```

aka make them clear groups. I can extend the change, if you want/agree.

Reply:
I actually find the more compact larger if-block you propose harder to read - but only slightly ^^
(I tend to avoid extremely long if-conditions if there is a better option, like bailing out early)

So, I have a slight preference for the current style, but that's it, I also see advantages in the compact version.
tl;dr: I wouldn't object the change, it's just a style choice and either is readable.

Reply:
Thanks :-) I had a slight preference for the existing style because it's used everywhere else for these functions as well, and I value consistency quite a lot. But yeah, minor thing :-)